test/e2e-common.sh

```diff
@@ -166,6 +172,32 @@ function install_istio() {
 
   echo ">> Running Istio"
   kubectl apply -f "${INSTALL_ISTIO_YAML}" || return 1
+
+  # Depending on the flag `HTTPS_ONLY`, reapply the gateway yaml with https only
+  if (( HTTPS_ONLY )); then
+    # Enable SDS at ingress gateway and deploy the ingress gateway agent
+    helm template install/kubernetes/helm/istio/ --name istio \
+      --namespace istio-system -x charts/gateways/templates/deployment.yaml \
+      --set gateways.istio-egressgateway.enabled=false \
+      --set gateways.istio-ingressgateway.sds.enabled=true > \
+    ${istio_base}/istio-ingressgateway.yaml
+    kubectl apply -f ${istio_base}/istio-ingressgateway.yaml
+
+    export SECURE_INGRESS_PORT=$(kubectl -n istio-system \
+      get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+    export INGRESS_HOST=$(kubectl -n istio-system \
+      get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+
+    # This script generates certificates for example.com
+    source generate.sh
+
+    # Create a secret for the ingress gateway.
+    # Note that the credential name needs to be the same as the one listed
+    # in `test/config/https-gateway/https-gateway.yaml`
+    kubectl create -n istio-system secret generic tls-credential \
```

Comment:
We may want to delete this created secret in `knative_teardown()` below? I guess that the test will fail due to `secrets already exists` when we repeated it.